### PR TITLE
Prevent imports when running validation

### PR DESCRIPTION
**Pull request by therve**
pylint by default builds the AST of all the dependencies it can find
when running the py3k validation. In some cases (kubernetes), it can
take a while to run, while not providing value as we don't care about
checking attributes from those. This works around it by setting sys.path
just when running the linter, so that imports fail and validation runs
faster.

### CODE REVIEW

**Review by ofek — APPROVED** [review]
Ha nice!

**Comment by therve** [comment]
I found out about this: https://github.com/PyCQA/pylint/blob/6b3afd4f6d75debd7f286f0d3c760ed10ab1e79f/pylint/checkers/python3.py#L693-L788

If we want those to work the latest change should be better.

File: a7/validate.py [hunk 1]
```diff
@@ -25,7 +26,15 @@ def validate_py3(path_to_module):
         # cf https://www.python.org/dev/peps/pep-0328/
         linter.disable("no-absolute-import")
         with fix_import_path([path_to_module]):
-            linter.check(path_to_module)
+            syspath = sys.path[:]
+            try:
+                # Remove site-packages from imports. It keeps the standard
+                # library, but it prevents pylint from parsing potentially big
+                # third party modules.
+                sys.path = [p for p in syspath if 'site-packages' not in p]
```
Comment (truthbk) [inline a7/validate.py:34]: I feel like this behavior should be behind a command flag.

Comment (therve) [reply a7/validate.py:34]: I hear you but If we want to get that in early, that means making an agent change to get the better performance.

And it **shouldn't** change behavior. If it does, we need to fix it, not have a flag to revert back to the old one.

**Review by ofek — APPROVED** [review]

**Pull request merged**